### DISCUSSION TITLE
Question about modeling data

### DISCUSSION THREAD

**Discussion by HBeserra** [prisma/prisma#12511] [Q&A]
Hi there!

I'm currently trying to model some data, but unsure of how to approach this:

I have a **User**, **Store**, **products** models, where users, products, and stores can have images, but the image has to have the relation only one of them. Because if the user(salesman) is removed, the product can't lose its image, but the profile picture have to be removed from the database. 

The image will be saved in a bucket with the uuid as the filename and once a mouth the Images that are not in the database will be removed from the bucket.

How the image can have  from a table can be referenced by one of multiple tables?

**Comment by nurul3101** [comment]
Hey @HBeserra 👋 
Are you planning to have a separate `Image` model which should contain images of User, store or product?

**Reply by HBeserra** [reply]
No, the same model, for all images to make all accessible from the same place.

**Reply by pantharshit00** [reply]
Hello @HBeserra, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

You might be able to use an enum here. 

```prisma
model Image {
  id String @id @default(cuid())
  type ImageType
  // ... other fields
}

model User {
  id String @id @default(cuid())
  type ImageType
  // ... other fields
}

// ...other Models

enum ImageType {
  PROFILE
  PRODUCT
 //....
}
```

Let me know if this addresses your usecase or you need something else.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @HBeserra , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Reply by HBeserra** [reply]
Hi @pantharshit00 and @vladi-stevanovic, thanks for the answer! I ended up choosing to use native calls to the database in this project.

Have a great day!

**Reply by vladi-stevanovic** [reply]
Thank you for letting us know! Don't hesitate to open a new Discussion if you have any other questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
